### PR TITLE
(SERVER-1118)Validate code id command

### PR DESCRIPTION
**Pull request by erikPrime**
fixed up teardown
removed unneeded comments at the end.

### CODE REVIEW

File: acceptance/suites/tests/code_commands/code_scripts.rb [hunk 1]
```diff
@@ -0,0 +1,181 @@
+require 'json'
+
+confine :except, :type => 'pe'
+
+test_name 'SERVER-1118: Validate code-id-command feature in FOSS'
+
+git_repo='/git/puppetcode' #AKA "Control Repo" &| "Git Server" &| "origin"
+git_local_repo='/tmp/git'
```
Comment (kurtwall) [inline acceptance/suites/tests/code_commands/code_scripts.rb]: Spaces around operators? Not real crazy about the comment at the end of the line, either.

File: acceptance/suites/tests/code_commands/code_scripts.rb [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  on(master, "/opt/puppetlabs/puppet/bin/gem uninstall r10k")
+  #return /etc/puppetlabs/code to original state
+  on(master, "rm -rf /etc/puppetlabs/code")
+  on(master, "puppet resource file /etc/puppetlabs/code ensure=directory")
+  #remove code_* scripts.
+  on(master, "rm -rf /opt/puppetlabs/server/apps/puppetserver/code-id-command_script.sh")
+  on(master, "rm -rf /opt/puppetlabs/server/apps/puppetserver/code_content_script.sh")
+end
+
+step "SETUP: Generate a new ssh key for the root user account to use with the git server"
```
Comment (kurtwall) [inline acceptance/suites/tests/code_commands/code_scripts.rb]: Nit: Like to see consistent use of single- and double-quotes. Most of the other `step` statements use single-quotes.

File: acceptance/suites/tests/code_commands/code_scripts.rb [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+###############################################################################
+step "Get the current code-id"
+  current_code_id=on(master, "/opt/puppetlabs/server/apps/puppetserver/code-id-command_script.sh production").stdout.chomp
+
+step "Pull the catalog, validate that it contains the current code-id"
+  cacert=   "/etc/puppetlabs/puppet/ssl/certs/ca.pem"
+  key=      "/etc/puppetlabs/puppet/ssl/private_keys/#{fqdn}.pem"
+  hostcert= "/etc/puppetlabs/puppet/ssl/certs/#{fqdn}.pem"
+  auth_str= "--cacert #{cacert} --key #{key} --cert #{hostcert}"
+  result=on(master, "curl --silent #{auth_str} https://#{fqdn}:8140/puppet/v3/catalog/#{fqdn}?environment=production | python -m json.tool").stdout
```
Comment (kurtwall) [inline acceptance/suites/tests/code_commands/code_scripts.rb]: Spaces around `=`?

File: acceptance/suites/tests/code_commands/code_scripts.rb [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+step 'SETUP: Initialize the local git repository'
+  on master, "mkdir #{git_local_repo}"
+  on master, "cd #{git_local_repo} && git init"
+  on master, "cd #{git_local_repo} && touch .gitignore"
+  on master, "cd #{git_local_repo} && git add ."
+  on master, "cd #{git_local_repo} && git commit -m 'initial commit'"
+  on master, "cd #{git_local_repo} && git remote add origin git@#{fqdn}:#{git_repo}"
+  on master, "cd #{git_local_repo} && git push origin master"
+
+################################################################################
```
Comment (kurtwall) [inline acceptance/suites/tests/code_commands/code_scripts.rb]: This seems out of place.

**Comment by kurtwall** [comment]
:see_no_evil:

**Comment by erikPrime** [comment]
This fails in ubuntu because of a problem with 'gem install r10k' on line 93.  Please don't merge yet.

**Comment by erikPrime** [comment]
Never mind; this was a temporary network problem that seems to have been resolved.

**Pull request merged**